src/herder/HerderSCPDriver.cpp

```diff
@@ -187,13 +218,43 @@ HerderSCPDriver::validateValueHelper(uint64_t slotIndex,
 
     bool compat = isSlotCompatibleWithCurrentState(slotIndex);
 
+    auto const& lcl = mLedgerManager.getLastClosedLedgerHeader().header;
+
+    // when checking close time, start with what we have locally
+    lastCloseTime = lcl.scpValue.closeTime;
+
     if (compat)
     {
-        lastCloseTime = mLedgerManager.getLastClosedLedgerHeader()
-                            .header.scpValue.closeTime;
+        if (!checkCloseTime(slotIndex, lastCloseTime, b))
+        {
+            return SCPDriver::kInvalidValue;
+        }
     }
     else
     {
+        if (slotIndex == lcl.ledgerSeq)
+        {
+            // previous ledger
+            if (b.closeTime != lastCloseTime)
+            {
+                return SCPDriver::kInvalidValue;
```

Comment:
Perhaps we should log in each case that we return without invoking `checkCloseTime`?

Reply:
fixed